### PR TITLE
[WNMGDS-2124] a11y footer refactor

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary

WNMGDS-2124

Refactor hgov and mgov footers to rely on `aria-hidden` span tags to act as delimiter. The way our footers are currently coded, we use `::after` and style this delimiter, but screen readers can read this content and it's bad UX.

Because the markup changed, I adjusted some of the CSS to better work with the new HTML.

## How to test

View both footers via demo site (SB hgov footer and mgov footer) or build locally and view

### CODE REVIEW

File: packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx [hunk 1]
```diff
@@ -8,24 +8,20 @@ interface InlineLinkListsProps {
...
-        </a>
-      ) : (
-        links[key]
-      );
+  return Object.getOwnPropertyNames(links).map(function (key, index) {
+    const link = typeof links[key] === 'string' ? <a href={links[key]}>{t(key)}</a> : links[key];
     return (
       <li key={key} className={inlineLiClasses}>
         {link}
+        {index !== Object.getOwnPropertyNames(links).length - 1 ? (
```
Comment (zarahzachz) [inline packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx:22]: Logic needed so span doesn't appear on last item in link list

File: packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx [hunk 2]
```diff
@@ -59,25 +55,27 @@ const InlineLinkLists = function (props: InlineLinkListsProps) {
 
   return (
     <div className="ds-l-container">
-      <ul className="ds-c-list ds-c-list--bare ds-u-font-size--small ds-u-margin-bottom--1">
+      <ul role="list" className="hc-c-footer__list ds-u-margin-bottom--1">
```
Comment (zarahzachz) [inline packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx:58]: since lists aren't styled, the `role="list"` was added to all ul tags

Comment (zarahzachz) [reply packages/ds-healthcare-gov/src/components/Footer/InlineLinkLists.tsx:58]: removed some repetitive util classes and moved into CSS file under `hc-c-footer` to reduce lines of code

File: packages/ds-healthcare-gov/src/styles/components/_Footer.scss [hunk 3]
```diff
@@ -3,9 +3,12 @@
...
 
 .hc-c-footer {
+  color: var(--color-muted);
+  font-size: var(--font-size-sm);
+
   a {
-    color: var(--color-muted);
-    text-decoration: none;
+    color: currentColor;
+    text-decoration-color: transparent;
```
Comment (zarahzachz) [inline packages/ds-healthcare-gov/src/styles/components/_Footer.scss:11]: `text-decoration-color` added to support HCM users

File: packages/ds-healthcare-gov/src/styles/components/_Footer.scss [hunk 4]
```diff
@@ -14,48 +17,41 @@
...
 }
 
-.hc-c-footer__inline-item {
-  color: var(--color-muted);
-  // We use a right margin and padding to avoid a weird
-  // left indent when an item wraps to the next line
-  margin-right: $spacer-half;
+.hc-c-footer__list {
+  align-items: center;
+  display: flex;
```
Comment (zarahzachz) [inline packages/ds-healthcare-gov/src/styles/components/_Footer.scss:22]: using flex on the ul/list itself helps with the weird spacing issues encountered when the links wrapped to new lines - removed padding/margin hack for flex.

**Review by pwolfert — APPROVED** [review]
I love that you made the CSS both more robust _and_ reduced the amount of it. The visual regression test changes look good 👍 . Great work!

**Comment by pwolfert** [comment]
Looks like it's failing CI due to unit test snapshot changes

**Pull request merged**